include/lsst/meas/modelfit/RegularizedMoments.h

```diff
@@ -0,0 +1,123 @@
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <https://www.gnu.org/licenses/>.
+ */
+
+#ifndef LSST_MEAS_MODELFIT_REGULARIZEDMOMENTS_H
+#define LSST_MEAS_MODELFIT_REGULARIZEDMOMENTS_H
+
+#include <Eigen/Dense>
+
+namespace lsst {
+namespace meas {
+namespace modelfit {
+
+/**
+ *  @brief This class is a model for calculating observed shape moments and their derivatives
+ *
+ *  This class models the biased weighted moments of an astronomical source given a vector of
+ *  moments intrinsic to an object, and a vector of moments representing a weighting function.
+ *
+ *  The class is constructed with the moments of weight function, and the moments of the object
+ *  are set by calling the at method. This allows a single instance to be reused to evaluate
+ *  multiple positions in object moment space. This structure is computationally more efficient
+ *  as well since the weight function would remain constant for many evaluation of image moments 
+ *
+ *  This class is designed to be used as an objective function in an optimizer which will find the
+ *  maximally likely image moments given moments measured from an image. As such the class
+ *  provides two methods in addition to the at method already mentioned. One method computes the
+ *  biased weighted moments given the specified weight and "true" moments, and the other computes
+ *  the derivative of the objective function along each of the image moment variables.
+ */
+class MomentsModel {
+public:
+    using Moments = Eigen::Matrix<double, 6, 1>;
+    using Jacobian = Eigen::Matrix<double, 6, 6>;
+    using FirstMoment = Eigen::Matrix<double, 2, 1>;
+    using SecondMoment = Eigen::Matrix<double, 2, 2>;
+
+    /**
+     * Construct a MomentsModel object with a given vector of weight function moments
+     *
+     * @param[in]   W       An eigen vector of the moments of the elliptical Gaussian weighting function.
+     *                      Most likely the first element (the zeroth moment) should be one, representing
+     *                      a normalized weight function.
+     */
+    MomentsModel(Moments const & W): W(W) {
+    }
+
+    /**
+     *  Sets the moments corresponding to the astronomical object's intrinsic moments
+     *
+     *  When this function is called, it sets the astronomical object's intrinsic moments for all subsequent
+     *  calculations. Additionally after setting the moment, this function calculates and caches the value
+     *  of the biased weighted moments given the weights and intrinsic moments.
+     *
+     *  @param[in]  Q       An eigen vector of the intrinsic moments of the astronomical object
+     */
+    void at(Moments const & Q);
```

Comment:
Name is confusing. Looks like `std::vector<T>::at` which is used for bounds-checked element access.
Also violates naming conventions. Perhaps use `setMoments` instead?

Reply:
Suggestion. If the object is to be used as a function, why not make it a functor?

Reply:
changed to setParameters